scripts/rosdistro_freeze_source

```diff
@@ -75,11 +77,18 @@ def main():
         args.dist_names = sorted(index.distributions.keys())
 
     for dist_name in args.dist_names:
+        dist_url = index.distributions[dist_name]['distribution']
+        if type(dist_url) == list:
+            if len(dist_url) != 1:
+                print("This script only works for distributions with a single distribution.yaml.", file=sys.stderr)
+                continue
+            dist_url = dist_url[0]
```

Comment:
I looked at the `Index()` class; I didn't see an obvious way to get this from there in an already version-generic way.